Aventra-MyEID-PKI-card.md

```diff
@@ -0,0 +1,150 @@
+
+```bash
+pkcs15-init -F
+```
+
+### Smart card reader configuration
+
+MyEID card uses T=1 protocol. This basically means that the response data is sent with the answer to the command/request. With T=0 protocol the smart card will first answer to the command and tell how much data it will send. Data is then requested separately.
+
+In some environments there has been issues when reading files that exceed some threshold. If you encounter problems when reading larger files from the card (e.g. certificates) with no apparent reason, try to set the readers `max_recv_size` (max receive size) to e.g. 192, to be on the safe side. You can then try to iterate to find the maximum for your environment.
+
+The setting in the `opensc.conf` (usually in `/etc` or `/etc/opensc`) config file is the following:
+
+```text
+...
+	reader_driver pcsc {
+		# This sets the maximum send and receive sizes.
+		# Some reader drivers have limitations, so you need
+		# to set these values. For usb devices check the
+		# properties with lsusb -vv for dwMaxIFSD
+		#
+		# max_send_size = 254;
+		# max_recv_size = 254;
+		max_recv_size = 192;
+...		
+	}
+
+	reader_driver openct {
+...
+		# max_send_size = 252;
+		# max_recv_size = 252;
+		max_recv_size = 192;
+...
+	};
+```
+
+## Links & other information
+
+Card details can found in [Reference manual](https://aventra.fi/wp-content/uploads/2024/03/MyEID-PKI-JavaCard-Applet-Reference-Manual.pdf).
+
+Cards can be bought from Aventra as blank cards or according to customer specifications regarding appearance etc. Small quantities of cards and readers can be easily bought from the web shop. For larger quantities contact Aventra sales for a quote.  
+
+* [Aventra Ltd.](https://aventra.fi/)
+* [Web shop](https://webservices.aventra.fi/webshop/)
+
+### About Aventra
+
+Aventra is a high tech company specializing in information security products and services. We are especially focusing on Public Key Infrastructure technologies. Most of our products are developed in house.
+
+Aventra offers a complete portfolio of card products ranging from simple plastic cards to high security smart cards and tokens. Our most recent product line features security solutions for mobile applications.  We also provide complete services and systems for issuing and managing cards and secure tokens, including card printers and materials.
+
+## Speed
+
+* RSA 1024 key generation: 0m8.280s
+* RSA 2048 key generation: 0m41.037s
+
+## Notes
+
+* Card requires a PUK code when creating a PIN code (fails to create a PIN without a PUK).
+* A minidriver is available for download [here](https://aventra.fi/downloads/).
+* ou can **not** upload custom Java-Applets like the openpgpcard-applet to the Aventra MyEID-card because the card is locked and Aventra refuses to hand out the required PIN.
```

Comment:
```suggestion
* You can **not** upload custom Java-Applets like the openpgpcard-applet to the Aventra MyEID-card because the card is locked and Aventra refuses to hand out the required PIN.
```